atsign-foundation/at_c
# fix: implment new method of trimming 'data:' from server responses

Closes https://github.com/atsign-foundation/at_c/issues/288

**- What I did**
- fix the issue: 'bytes read had some leading excess bytes'

**- How I did it**
- Introduce a method in string_utils.c named "atclient_string_utils_get_substring_position()" that locates the position of a given substring.
- Using the above method to locate the 'data:' substring and then moving the pointer 5 positions ahead to skip the 'data:' substring.
- Replaced existing method of trimming 'data:' prefix if a server response starts with 'data:', with the new implementation

**- How to verify it**
- Successful build should indicate no compile-time errors. Yet to do a manual test to verify individual functionalities updated with changes in this PR.

**- Description for the changelog**

fix: implment new method of trimming 'data:' from server responses
